File: call-in-script/do-loops.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-
import os
import glob
import subprocess

os.putenv("SAC_DISPLAY_COPYRIGHT", "0")

p = subprocess.Popen(['sac'], stdin=subprocess.PIPE)

s = ""
s += "echo on \n"
for filename in glob.glob("*.SAC"):
    s += "r %s \n" % (filename)
    s += "rmean; rtr; taper \n"
    s += "lp co 1.0 p 2 n 4 \n"
    s += "w %s.filtered\n" % (filename)
s += "quit \n"

p.communicate(s.encode())

File: call-in-script/do-loops.pl
#!/usr/bin/env perl
use strict;
use warnings;

$ENV{SAC_DISPLAY_COPYRIGHT}=0;
open(SAC, "| sac ") or die "Error opening sac \n";
print SAC "echo on \n";
foreach (glob("*.SAC")) {
    print SAC "r more $_ \n";
}
print SAC "rmean \n";
print SAC "rtrend \n";
print SAC "lp co 1.0 p 2 n 4 \n";
print SAC "write append .filtered \n";
print SAC "quit \n";
close(SAC);
